### ISSUE TITLE
julia.exe starting error with Korean path-name.

### ISSUE THREAD

**Issue by nightlydebian** [JuliaLang/julia#33486]
Hello, everyone.
My setting is Julia 1.2 on Win10 with Korean locale.

After installing julia, everytime I try to start julia.exe, 
it is raising a error "sys.dll not found".

It seems that Julia.exe does not recognize the file-path including Korean characters(in my case, it is username). If the username consists with only ascii characters, there is no problem.  
In Julia REPL itself, everything is fine with Korean file-path.

I hope this report contributes to the Julia project. 
Thank you!

**Comment by wookay** [comment]
it's not working well which has installed on the destination path that contains Korean characters (it has not encoded as utf-8 by the historical reasons) in windows system.
installing the other programming languages (like Python, R, Matlab) got the same problem.

**Comment by StefanKarpinski** [comment]
Can you give a bit more context? What is the full path of the `sys.dll` library?

**Comment by wookay** [comment]
some tests with `C:\테스트` (copied the installed directory from C:\julia)

```
C:\테스트\bin> julia
ERROR: could not load library "C:\테스트\lib\julia\sys.dll"
The specified module could not be found.

C:\테스트\bin> julia -v
julia version 1.2.0

C:\테스트\bin> julia -H C:\julia\bin -E "1"
1
```

**Comment by wookay** [comment]
CI works fine without `sys.dll` error.
 - travis-ci https://travis-ci.org/wookay/Test33486/builds/595229292#L49
 - appveyor https://ci.appveyor.com/project/wookay/test33486#L27

**Comment by StefanKarpinski** [comment]
That appears to be a valid Unicode string. Is there some encoding mismatch happening?

**Comment by wookay** [comment]
when user name or copy of the directory with having the Korean characters in windows, 
the name of the directory (`C:\테스트`) itself has encoded by [cp949](https://en.wikipedia.org/wiki/Unified_Hangul_Code).

* run on [ConEmu](https://conemu.github.io/), installed [iconv](https://mlocati.github.io/articles/gettext-iconv-windows.html) in windows 8.
```
$ chcp 65001
$ echo mkdir c:\테스트\lib\julia | iconv -f utf-8 -t cp949 | clip
# <pasted the cp949 encoded mkdir command>
$ echo copy c:\julia\lib\julia\sys.dll c:\테스트\lib\julia\ | iconv -f utf-8 -t cp949 | clip
# <pasted the cp949 encoded copy command>
```

now we got the `C:\�׽�Ʈ\lib\julia\sys.dll`. (`�׽�Ʈ` has broken to read) and can run this
```
C:\>cd C:\테스트\bin

C:\테스트\bin>julia.exe -E "1"
1

C:\테스트\bin> C:\테스트\bin\julia.exe -E "1"
1

C:\> cd \
C:\> c:\테스트\bin\julia.exe -H C:\�׽�Ʈ\bin -E "1"
1
```

copied the directories in the CI (travis, appveyor), they have encoded as utf-8.

**Comment by StefanKarpinski** [comment]
I wonder if this has to do with libuv's refusal to deal with invalid UTF-16 data. @vtjnash? It seems like if they were using WTF-8 as we've recommended, then we would give Windows back whatever it considers the directory name to be (regardless of what the encoding is) and it would find it.

**Comment by ViralBShah** [comment]
This seems to work fine on mac, but not sure about windows (and hence suggesting it may not be a libuv issue). If it is still a problem, can you please reopen this?

**Comment by StefanKarpinski** [comment]
I'm pretty sure the issue here is that Windows allows invalid UTF-16 in path names (any maybe telling it to use cp949 causes that to happen?) but libuv chokes on it. I made a [pull request to libuv](https://github.com/libuv/libuv/pull/2970) to change it to use WTF-8, which allows faithfully representing any invalid UTF-16 as well-formed UTF-8 (with code points that you're not "supposed" to use). This was somewhat contentious, but is IMO clearly the right thing to do here. That's stalled because I don't have access to a Windows system to test on and am not familiar enough with libuv's testing to write tests to make sure that it works correctly. It would be great if someone finished and merged that pull request, but I'm clearly not going to do it. Perhaps @vtjnash could do so.

**Comment by ViralBShah** [comment]
Seems like we need to leave this open.

**Comment by giordano** [comment]
For what is worth, in https://github.com/JuliaGraphics/Gtk.jl/issues/461 we found that non-ASCII characters in `PATH` on Windows wreak total havoc.  I wouldn't be surprised if this was a similar issue.

**Comment by StefanKarpinski** [comment]
Yeah libuv just can't handle it. Julia is fine with it. So if libuv produces WTF-8 it should just work. Or at least if something doesn't work it will be easy to fix.

**Comment by inkydragon** [comment]
libuv/libuv#2970 is merged. libuv v1.46.0 include it.
We are using libuv-uv2-1.44.2, so Do we have plan to update libuv? @vtjnash

**Comment by vtjnash** [comment]
Yes. Started as https://github.com/JuliaLang/julia/pull/49937, but there is another libuv release planned soon (https://github.com/libuv/libuv/issues/4248) so I was wait for that

**Comment by inkydragon** [comment]
Reopen: After testing on v1.11.0-beta1, this issue has not been fixed.

https://discourse.julialang.org/t/julia-install-issue-foreign-name/114522/10

**Comment by StefanKarpinski** [comment]
It would be very funny if `MultiByteToWideChar` chokes on WTF-8 that was generated because Windows gives us invalid UTF-16. If that's the case, then we may have to do this translation ourselves.

**Comment by StefanKarpinski** [comment]
Indeed, that looks to be the case: https://internals.rust-lang.org/t/osstr-wtf8-as-bytes-and-to-string-unchecked/12694/13. There's an option (see the following post) but the options are "error on invalid UTF-8 or replace invalid characters with U+FFFD, neither of which is what we want—namely "handle WTF-8 in the natural way, but converting it to (invalid) UTF-16".

**Comment by vtjnash** [comment]
All the utility functions for it are exported from libuv, since Windows will indeed choke on them. It would be a minor project for someone to convert our uses of the Windows API calls to use the libuv ones instead.
